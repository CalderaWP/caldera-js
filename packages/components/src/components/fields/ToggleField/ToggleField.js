import {
	parseAttributes,
	fieldClassNames,
	isValidHtml5type,
	toBoolean
} from '../util';

import PropTypes from 'prop-types';
import React from 'react';
import { BaseControl, ToggleControl } from '@wordpress/components';
export const ToggleField = ({
	label,
	description,
	fieldId,
	placeholder,
	required,
	html5type,
	onChange,
	onBlur,

								checked,
}) => {

	return(
		<BaseControl
			id={fieldId}
			label={label}
			help={description}
		>
			<ToggleControl
				onBlur={onBlur}
				checked={ checked }
				onChange={ onChange }
			/>
		</BaseControl>

	)
};

ToggleField.propTypes = {
	label: PropTypes.string,
	description: PropTypes.string,
	fieldId: PropTypes.string,
	placeholder: PropTypes.string,
	required: PropTypes.bool,
	checked: PropTypes.bool,
	onChange: PropTypes.func,
	onBlur: PropTypes.func
};

ToggleField.defaultProps = {
	onBlur: () => {},
	required: false,
};
