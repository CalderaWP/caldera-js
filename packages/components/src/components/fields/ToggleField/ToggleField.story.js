import React from 'react';
import { storiesOf } from '@storybook/react';
import { ToggleField } from './ToggleField';

let value = 'Hi Roy';
const onChange = newValue => {
	console.log(newValue);
};
const onBlur = event => {};


storiesOf('ToggleField', module).add('Checked', () => (
	<ToggleField
		label={'Toggle field'}
		onChange={onChange}
		onBlur={onBlur}
		checked={true}
	/>
));

storiesOf('ToggleField', module).add('Not Checked', () => (
	<ToggleField
		label={'Toggle field'}
		onChange={onChange}
		onBlur={onBlur}
		help={'This One Is Not Checked'}
		checked={false}
	/>
));
