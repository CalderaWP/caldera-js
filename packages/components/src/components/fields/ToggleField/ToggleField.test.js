import * as React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { ToggleField } from './ToggleField';

describe('ToggleField ', () => {
	let onChange;
	let onBlur;

	beforeEach(() => {
		onChange = jest.fn();
		onBlur = jest.fn();
	});

	it('matches snapshot with all props', () => {
		const component = renderer.create(
			<ToggleField
				label={'Hi Roy'}
				description={'Say Hi'}
				value={'Roy'}
				onChange={onChange}
				onBlur={onBlur}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Changes calls change handler', () => {
		const component = mount(
			<ToggleField
				label={'Hi Roy'}
				description={'Say Hi'}
				placeholder={'Hello'}
				html5type={'email'}
				value={'Roy'}
				onChange={onChange}
				onBlur={onBlur}
			/>
		);

		component.find('input').simulate('change');
		expect(onChange.mock.calls.length).toBe(1);
	});
});
