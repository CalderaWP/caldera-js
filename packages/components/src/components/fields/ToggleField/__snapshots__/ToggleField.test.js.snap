// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleField  matches snapshot with all props 1`] = `
<div
  className="components-base-control"
>
  <div
    className="components-base-control__field"
  >
    <span
      className="components-base-control__label"
    >
      Hi Roy
    </span>
    <div
      className="components-base-control components-toggle-control"
    >
      <div
        className="components-base-control__field"
      >
        <span
          className="components-form-toggle"
        >
          <input
            className="components-form-toggle__input"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="components-form-toggle__track"
          />
          <span
            className="components-form-toggle__thumb"
          />
          <svg
            aria-hidden="true"
            className="components-form-toggle__off"
            focusable="false"
            height="6"
            role="img"
            viewBox="0 0 6 6"
            width="6"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 1.5c.8 0 1.5.7 1.5 1.5S3.8 4.5 3 4.5 1.5 3.8 1.5 3 2.2 1.5 3 1.5M3 0C1.3 0 0 1.3 0 3s1.3 3 3 3 3-1.3 3-3-1.3-3-3-3z"
            />
          </svg>
        </span>
        <label
          className="components-toggle-control__label"
        />
      </div>
    </div>
  </div>
  <p
    className="components-base-control__help"
    id="undefined__help"
  >
    Say Hi
  </p>
</div>
`;
